### PR TITLE
DAOS-12064 test: Remove HW Small stage from pipeline-lib testing

### PR DESCRIPTION
**Pull request by phender**
Update pipeline-lib dependency building stages to remove the no longer used Functional Hardware Small stage.

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

### CODE REVIEW

File: vars/packageBuildingPipelineDAOSTest.groovy [hunk 1]
```diff
@@ -755,8 +755,6 @@ void call(Map pipeline_args) {
                                                    booleanParam(name: 'CI_SCAN_RPMS_el7_TEST', value: false),
                                                    booleanParam(name: 'CI_RPMS_el7_TEST',
                                                                 value: 'centos7' in distros),
-                                                   booleanParam(name: 'CI_small_TEST',
-                                                                value: 'el8' in distros),
```
Comment (brianjmurrell) [inline vars/packageBuildingPipelineDAOSTest.groovy:759]: This is removing all hardware testing from dependency packages.  We should run at least one hardware stage on them.  It seems it should be medium now though.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: vars/packageBuildingPipelineDAOSTest.groovy (hunk 1 above)
Comment (phender) [reply vars/packageBuildingPipelineDAOSTest.groovy]: Fixed

**Review by brianjmurrell — APPROVED** [review]

**Comment by phender** [comment]
In https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fpipeline-lib/detail/PR-351/4/pipeline/1167/ the master branch testing in https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/ci-daos-stack-pipeline-lib-PR-351-master/8/pipeline/ was marked with a red X, but all the stages are green and there are no test failures.

**Comment by brianjmurrell** [comment]
> In https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fpipeline-lib/detail/PR-351/4/pipeline/1167/ the master branch testing in https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/ci-daos-stack-pipeline-lib-PR-351-master/8/pipeline/ was marked with a red X, but all the stages are green and there are no test failures.

This was due to https://github.com/daos-stack/daos/commit/ff905beca5daf5bd8c21f7898b72aa6e3fe14698 which adds an empty valgrind stash if the _Fault injection testing on EL 8_ stage does not cause any segfaults (as that stage's test only runs with valgrind in a second pass if a segfault is found in a first pass).

The code to unstash valgrind stashes https://github.com/daos-stack/pipeline-lib/blob/8345c86d07808b3d186718994a57ebf8ac728f45/vars/valgrindReportPublish.groovy#L36-L49 handles _missing_ stashes (i.e. for when a stage is skipped) but does not handle empty stashes and so ends up calling https://github.com/daos-stack/pipeline-lib/blob/8345c86d07808b3d186718994a57ebf8ac728f45/vars/valgrindReportPublish.groovy#L55-L66 with no valgrind files if all of the other stages that normally produce valgrind files are skipped.  This is an error for [`publishValgrind()`](https://www.jenkins.io/doc/pipeline/steps/valgrind/#publishvalgrind-publish-valgrind-reports).

#352 adds code to handle this situation as there is no way to handle it using Jenkins steps (i.e. either allowing no valgrind files, or [not creating a stash if there are actually no files to stash](https://issues.jenkins.io/browse/JENKINS-70211)).

**Pull request merged**